onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.h

```diff
@@ -99,7 +99,7 @@ struct TensorrtFuncState {
   std::string trt_node_name_with_precision;
   bool engine_cache_enable;
   std::string engine_cache_path;
-  nvinfer1::IRuntime* runtime = nullptr;
+  tensorrt_ptr::unique_pointer<nvinfer1::IRuntime>* runtime = nullptr;
```

Comment:
raw pointer to unique_ptr is kind of strange.
would it be better to just keep it as a raw pointer here?
i.e. the nvinfer1::Runtime object is managed by unique_ptr as part of TensorRTExecutionProvider ?

Reply:
Explained in other comment.